test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/schema/SchemaEntry.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.services.test.utils.core.schema;
+
+import java.net.URL;
+
+public class SchemaEntry {
+
+    private URL url = null;
+
+    private String filename = null;
+
+    private String checksum = null;
+
+    public SchemaEntry(URL url, String filename) {
```

Comment:
File can have url, and filename can be extracted from url, therefore only 1 parameter is enough here.

Reply:
you can do that only with the usage of helper methods. this makes it much simpler at a low memory penalty.

Reply:
* Again this risks a SchemaEntry instance to be inconsistent.
* No new helper method is needed. It's a simple `url.getPath()` or `url.getFile()`
* Performance wise, without the extra `filename` field, an instance will take 4 or 8 less bytes in heap space and GC will face a slightly less complicated object graph.